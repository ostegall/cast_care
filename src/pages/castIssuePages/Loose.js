import { Text, View, Pressable, Linking } from 'react-native';
import styles from '../../styles';
let phone = " (984)-974-5700";
let phone_url = "tel://+19849745700";

export default function Loose({ navigation }) {
  //Solutions for user that has loose cast
  return (
    <View style={{ flex: 1, paddingTop: 20 }}>
      <Text style={styles.title_text}>
        Problem Description:
      </Text>
      <Text style={styles.page_text}>
        There will be a little bit of wiggle room in the cast, anything beyond 2 fingers between patient's cast and patient's skin is too much.
      </Text>
      <Text style={styles.title_text}>
        Why this is a Problem:
      </Text>
      <Text style={styles.page_text}>
        This could indicate that the cast/splint is no longer supporting the fracture.
      </Text>
      <Text style={styles.title_text}>
        Correcting the Problem:
      </Text>
      <Text style={styles.page_text}>
        It is important to have a proper fitting cast to assist the healing process.
      </Text>
      <Text style={styles.title_text}>
        Who to Contact:
      </Text>
      <Text style={styles.page_text}>
        Schedule an appointment in the next 2 business days by calling 
        <Text style={styles.url_text} //separate text box for an inline link to call UNC hospital
          onPress={() => Linking.openURL(phone_url)}>
          {phone}
        </Text>
        . Special consideration: If patient's cast is likely to come off soon, you need to be seen on the same day or come into the Orthonow walk-in clinic.
      </Text>
      <View style={{ alignItems: 'center', justifyContent: 'center' }}>
        <View style={styles.button_row}>
          <Pressable style={styles.button}
            onPress={() => {
              navigation.navigate("Info") //Navigates to page for finding nearest hospital
            }}
          >
            <Text style={styles.button_text}>
              Find your nearest location
            </Text>
          </Pressable>
          <Pressable style={styles.button}
            onPress={() => {
              navigation.navigate("Loose Blur") //Navigates to blurred image of problem
            }}
          >
            <Text style={styles.button_text}>
              Show image of loose cast
            </Text>
          </Pressable>
        </View>
      </View>
    </View>
  )
}