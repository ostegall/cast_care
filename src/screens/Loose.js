import React from 'react';
import { Text, View } from 'react-native';

class Loose extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <View>
        <Text>Too loose? Selection
        </Text>
      </View>
    )
  }
}

export default Loose

// export default function NewScreen () {
  
// }