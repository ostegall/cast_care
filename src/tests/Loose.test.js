import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import Loose from '../pages/castIssuePages/Loose';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<Loose />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<Loose />).toJSON();
  expect(tree).toMatchSnapshot();
});