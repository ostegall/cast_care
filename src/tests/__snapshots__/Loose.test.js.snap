// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Problem Description:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    There will be a little bit of wiggle room in the cast, anything beyond 2 fingers between patient's cast and patient's skin is too much.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Why this is a Problem:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    This could indicate that the cast/splint is no longer supporting the fracture.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Correcting the Problem:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    It is important to have a proper fitting cast to assist the healing process.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Who to Contact:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Schedule an appointment in the next 2 business days by calling
    <Text
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "color": "blue",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
       (984)-974-5700
    </Text>
    . Special consideration: If patient's cast is likely to come off soon, you need to be seen on the same day or come into the Orthonow walk-in clinic.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "height": 83,
          "marginLeft": 24,
          "marginRight": 23,
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4292c7",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          Find your nearest location
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4292c7",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          Show image of loose cast
        </Text>
      </View>
    </View>
  </View>
</View>
`;
